pkg/metrics/service.go

```diff
@@ -152,11 +152,14 @@ func ConfigureMetrics(ctx context.Context, userMetricsConfig metricsConfig) erro
 	if err != nil {
 		return err
 	}
-	res := int32(p)
-	s, svcerr := GenerateService(res, "metrics")
```

Comment:
The service should always be generated. The service monitor should be a separate block